### DISCUSSION TITLE
Cache working inconsistently

### DISCUSSION THREAD

**Discussion by lobis** [community/community#26306] [Actions]
<p>Hello,</p>
<p>I have recently started moving our proyect CI from GitLab to GitHub Actions and I am struggling with the Cache action.</p>
<p>Here you can see the workflow file: <a href="https://github.com/rest-for-physics/restG4/actions/runs/2488111253/workflow" class="inline-onebox" rel="noopener nofollow ugc">fixed bug in validation · rest-for-physics/restG4@ad10afd · GitHub</a></p>
<p>The problem I am seeing is that cache seems to stop working randomly, giving various errors. Earlier today for example it started giving a <code>Warning: getCacheEntry failed: Cache service responded with 503</code> (<a href="https://github.com/rest-for-physics/restG4/runs/6860746398?check_suite_focus=true#step:4:12" class="inline-onebox" rel="noopener nofollow ugc">Use latest dependencies in pipeline image · rest-for-physics/restG4@ad10afd · GitHub</a>) even when reruning actions that had successuly executed the day before.</p>
<p>This error 503 makes the action fail when retrieving the cache, and sometimes its also present when writing the cache, but its not always the case. In fact, here <a href="https://github.com/rest-for-physics/restG4/runs/6860746398?check_suite_focus=true#step:4:12" class="inline-onebox" rel="noopener nofollow ugc">Use latest dependencies in pipeline image · rest-for-physics/restG4@ad10afd · GitHub</a> it fails and here <a href="https://github.com/rest-for-physics/restG4/runs/6860708459?check_suite_focus=true#step:4:18" class="inline-onebox" rel="noopener nofollow ugc">Use latest dependencies in pipeline image · rest-for-physics/restG4@ad10afd · GitHub</a> it succeeds when retrieving the same cache entry on the same workflow run.</p>
<p>Also it happens that on some job it (apparently) correctly restores the cache but doesn’t return any error, it just doesn’t find the file, such as here: <a href="https://github.com/rest-for-physics/restG4/runs/6860708459?check_suite_focus=true#step:4:18" class="inline-onebox" rel="noopener nofollow ugc">Use latest dependencies in pipeline image · rest-for-physics/restG4@ad10afd · GitHub</a>. Again on the same workflow there are other jobs restoring the same cache key and finding that file such as <a href="https://github.com/rest-for-physics/restG4/runs/6860804655?check_suite_focus=true" class="inline-onebox" rel="noopener nofollow ugc">Use latest dependencies in pipeline image · rest-for-physics/restG4@ad10afd · GitHub</a> in this case.</p>
<p>I am not sure how to proceed, previously these errors happened in all of the jobs but after a few hours they began working again, now they only fail in some select jobs… I guess we are not using the cache action correctly but I don’t see how.</p>
<p>Thanks!</p>
<p>Edit: the cache size is always roughly 15Mb in size.</p>

**Comment by nik-humphries** [comment, marked as answer]
<aside class="quote" data-post="9" data-topic="256066">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/my-workflow-says-that-github-actions-is-down-but-the-status-is-green/256066/9">My workflow says that github actions is down but the status is green</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Hi there, 
Quick update! There is a minor interruption in the service and our team is working on it. We are sorry for the inconvenience. 
You may check the current status or subscribe for the incident on <a href="https://www.githubstatus.com/" rel="noopener nofollow ugc">https://www.githubstatus.com/</a>
  </blockquote>
</aside>

<p>There’s an update here if it’s the same as us. Acknowledged as an issue now</p>
